### PR TITLE
Forward Component

### LINKED ISSUE #499: [FEATURE] Forward component
**Describe the solution you'd like**
Create a Forward component based on the original Bootstrap Italia Forward component:
https://italia.github.io/bootstrap-italia/docs/componenti/forward/

* [ ] Create the basic component
* [ ] Add the logic to scroll
* [ ] Create a Storybook set of stories based on the BI documentation

*Notes*:
* use BI classes to style the component.
* no additional CSS is expected in this component, unless justified shortcomings of the original CSS.

### PR DESCRIPTION
**Pull request by ShreyansDubey**
Fixes #

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds a forward component to automatically scroll to any element as well as stories for it
closes #499
#### Changes proposed in this pull request:

-
-
-

### CODE REVIEW

**Comment by ShreyansDubey** [comment]
@dej611 Please review and suggest changes required (if any).

File: src/components/Forward/Forward.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ])
+}
+
+const Forward = ({ className, scrollToRef, ...attributes }) => {
+  const classes = classNames(className, 'forward')
+  return (
+    <a
+      className={classes}
+      onClick={() =>
+        scrollToRef.current.scrollIntoView({
```
Comment (dej611) [inline src/components/Forward/Forward.js]: What happens when `scrollToRef` is a function here?

Comment (ShreyansDubey) [reply src/components/Forward/Forward.js]: scrollToRef will be a callback function for a ref.
https://reactjs.org/docs/refs-and-the-dom.html#callback-refs

Comment (dej611) [reply src/components/Forward/Forward.js]: If I understood it correctly, that's a callback to get the `ref`.
Here instead you are using the `ref` to scroll the view to it: of the two `PropTypes` definitions for it, I see only the `shape` one that fits this code. If I pass a callback to this component as `scrollToRef` the `onClick` function I guess wilol throw and error, no?

Comment (ShreyansDubey) [reply src/components/Forward/Forward.js]: Ah yes it doesn't work, it doesn't throw an error as such but the callback ref should be handled differently

File: src/components/Forward/Forward.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          behavior: 'smooth',
```
Comment (dej611) [inline src/components/Forward/Forward.js]: Mind that this is not supported by IE (ok ,fair) and `Edge` pre-chromium version: https://caniuse.com/#search=scrollIntoView .
Is it ok @francescozaia ?

Comment (francescozaia) [reply src/components/Forward/Forward.js]: Actually, we're still supporting IE11 and early versions of Edge, as [they still sum up to a ~ 6% - 7% of usage in Italy](https://gs.statcounter.com/browser-version-market-share/all/italy), especially among public administrations.
So, I 'm sorry to be a pain, but I would use something else (or just some graceful degradation positioning for those browsers).

Comment (ShreyansDubey) [reply src/components/Forward/Forward.js]: From what I can see here, scrollIntoView is partially supported by IE11 and Edge pre-chromium, the part that is not supported is the smooth behaviour.

Comment (dej611) [reply src/components/Forward/Forward.js]: If the animation type is not a concern then it's ok.

Comment (ShreyansDubey) [reply src/components/Forward/Forward.js]: I checked this on virtualbox, for both Internet explorer 11 and pre-chromium edge. Scrolling works fine, but happens instantly, i.e. without any animation with the existing `scrollIntoView`

Comment (francescozaia) [reply src/components/Forward/Forward.js]: Great then! Good for me, thanks @Shreyans for checking! 👍

**Comment by ShreyansDubey** [comment]
@dej611 , @francescozaia Can this merged if there are no changes?

**Pull request merged**